tfrecord/reader.py

```diff
@@ -121,12 +123,39 @@ def tfrecord_loader(data_path: str,
         count. Necessary to evenly split/shard the dataset among many
         workers (i.e. >1).
 
+    transform_func : a callable, default = None
+        A function that takes in the input `features` i.e the dict
+        provided in the description and returns a desirable output.
+        It's useful for having transformation on our input to get
+        desired output.
+
+    removed_fields : list of str, default=None
+        One of the list of keys that we extract from each record
+        but don't want to return it.
+
```

Comment:
Remove remove_fields. You can simply not specify features that you don't want loaded.

Reply:
We have a use case here that where we just load a feature, perform a transformation but not necessarily want it to be yielded as torch Tensor. 

For simplicity imagine you have a feature called `file_path`.
Inside your `transform` you load the file at that path and return it with the name `image`.
Now the `file_path` is redundant and not required anymore.

Lmk if you still have an objection, I can remove it. 

Another place to look at could be here in [petastorms API](https://github.com/uber/petastorm#pytorch-api), they have something called `removed_fields`.

Reply:
You can simply not return it in transform. Let's remove iit.

Reply:
I'll remove it, just to confirm though, you're saying that the user should be responsible for explicitly removing any keys in `transform` themselves that they don't need anymore?

I believe you're suggesting something like the following

```python
description = {"file_index" = "int"}
def transformation(features):
        features["image"] = img = cv2.imread(f'{features["file_index"]}.png')
        features.pop("file_index", None)   
        return features

dataset = tfrecord.torch.TFRecordDataset("/path/to/data.tfrecord",
                                         index_path=None,
                                         description=description,
                                         transform=transformation)
```

instead of having removed fields?